src/Spice86.Core/Emulator/Devices/Sound/PcSpeaker.cs

```diff
@@ -292,8 +292,10 @@ private void Dispose(bool disposing) {
         }
 
         if (disposing) {
+            // Unregister callback from channel
+            _soundChannel.SetRenderCallback(null);
```

Comment:
you don't need to set to null an action, or implement Dispose for that. This is not C++